tests/modules/nf-core/sgdemux/main.nf

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { SGDEMUX } from '../../../../modules/nf-core/sgdemux/main.nf'
+
+workflow test_sgdemux {
+    
+    // input = Channel.value([
+    //    [ id: 'sim-data' ], // meta map
+    //    file("/Users/.../projects/SINGULAR/singular-demux/simulated/out.sample_meta.csv", checkIfExists: true),
+    //    file("/Users/.../projects/SINGULAR/singular-demux/simulated/", checkIfExists: true)
```

Comment:
TODO: add a tarball to `nf-core/test-datasets`, then have a process to untar it prior to running SGDEMUX, similar to: https://github.com/nf-core/modules/blob/master/tests/modules/nf-core/bases2fastq/main.nf